### PR TITLE
Cleanup/rewrite plrustc main.rs

### PR DESCRIPTION
**Pull request by thomcc**
Outside of work, I happened to write another rustc_driver wrapper. I managed to make it significantly simpler than what we're doing in plrustc, so here's a commit using that approach instead.

### CODE REVIEW

**Review by eeeebbbbrrrr — APPROVED** [review]
Definitely looks less complex.  I can't really review this for correctness, so assuming CI passes, go ahead and merge.

File: plrustc/plrustc/src/main.rs [hunk 1]
```diff
@@ -51,66 +36,24 @@ impl rustc_driver::Callbacks for PlrustcCallbacks {
 }
 
 fn main() {
-    // Snapshot the environment before we call functions which may mess with it.
-    // let env_snapshot = env::vars_os().collect::<Vec<_>>();
     rustc_driver::install_ice_hook();
```
Comment (workingjubilee) [inline plrustc/plrustc/src/main.rs:39]: :ice_cube:

File: plrustc/plrustc/src/main.rs [hunk 2]
```diff
@@ -127,124 +70,101 @@ fn main() {
...
-    // Invoke compiler, and handle return code.
-    let exit_code = rustc_driver::catch_with_exit_code(move || {
+fn run_compiler<CB: Callbacks + Send>(mut args: Vec<String>, callbacks: &mut CB) -> ! {
+    args.splice(1..1, std::iter::once("--cfg=plrustc".to_string()));
+    std::process::exit(rustc_driver::catch_with_exit_code(move || {
         rustc_driver::RunCompiler::new(&args, callbacks).run()
-    });
-    std::process::exit(exit_code)
+    }));
 }
```
Comment (workingjubilee) [inline plrustc/plrustc/src/main.rs:170]: Nice. -80 lines.

**Review by workingjubilee — APPROVED** [review]

File: plrustc/plrustc/src/main.rs (hunk 2 above)
Comment (thomcc) [reply plrustc/plrustc/src/main.rs:170]: Also a lot simpler conceptually. It no longer invokes itself!

**Pull request merged**